airflow/jobs.py

```diff
@@ -574,6 +547,10 @@ def __init__(
 
         self.heartrate = conf.getint('scheduler', 'SCHEDULER_HEARTBEAT_SEC')
         self.max_threads = conf.getint('scheduler', 'max_threads')
+
+        if log:
```

Comment:
Why is this required?

Reply:
Because the log is only overridden when it is launched from the DagProcessor. We could move it into the Scheduler, but that would mean the handler would always kick in. Maybe not what we want.